rust/src/main.rs

```diff
@@ -45,15 +46,7 @@ fn get_logging_directory() -> Result<PathBuf, io::Error> {
     }
 }
 
-fn path_for_log_file<P: AsRef<Path>>(filename: P) -> Result<PathBuf, io::Error> {
-    let mut logging_directory = get_logging_directory()?;
-    // Create the logging directory
-    fs::create_dir_all(&logging_directory)?;
-    logging_directory.push(filename.as_ref());
-    Ok(logging_directory)
-}
-
-fn setup_logging() -> Result<(), fern::InitError> {
+fn setup_logging(logging_path_result: Result<PathBuf, io::Error>) -> Result<(), fern::InitError> {
```

Comment:
I'm pretty sure the path should be completely known by this point, which means that this fn should take an Option<&Path> instead of a Result

Reply:
The reason I send through the PathBuf or the io:Error is the Error can be logged cleanly with fern on line 99 but I can remove it & do a normal eprintln if that's preferred.

Reply:
Ok, I'll go swap it to an `eprintln` & `Option`

Reply:
Done

Reply:
I was thinking that the caller could hold on to the Err and log it *after* calling this function.

Reply:
Ahh that would be better.

Reply:
Sorted